airflow/www/views.py

```diff
@@ -520,16 +521,60 @@ def last_dagruns(self, session=None):
     @has_access
     @provide_session
     def code(self, session=None):
-        dm = models.DagModel
-        dag_id = request.args.get('dag_id')
-        dag = session.query(dm).filter(dm.dag_id == dag_id).first()
-        try:
+
+        def get_serialized_dag(dag_id):
```

Comment:
These functions look very complex. Can you move them to another module? This will allow their use also by other classes, e.g. API

Reply:
Done